tests/vm_test.go

```diff
@@ -1921,6 +1921,56 @@ status:
 			}, 2*time.Minute, 1*time.Second).Should(Succeed())
 		})
 	})
+
+	Context("[Serial] when node becomes unhealthy", Serial, func() {
+		var nodeName string
+
+		AfterEach(func() {
+			libnode.Untaint(nodeName, libnode.UnschedulableTaint)
+		})
+
+		It("[Serial] the VMs running in that node should be respawned", func() {
+			By("Starting VM")
+			// We need to set RunningStrategy to either Always or ReRunOnFailure, the startVM function sets
+			// the RunningStrategy to Always.
+			vm := startVM(virtClient, createVM(virtClient, libvmi.NewAlpine()))
+
+			vmi, err := virtClient.VirtualMachineInstance(vm.Namespace).Get(context.Background(), vm.Name, &k8smetav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			oldUID := vmi.UID
+
+			By("Simulating the node becomes unhealthy")
+			nodeName = vmi.Status.NodeName
+			libnode.SimulateUnhealthyNode(virtClient, nodeName)
+
+			By("VMI should be NotReady")
+			Eventually(ThisVMI(vmi), 60*time.Second, 2*time.Second).Should(HaveConditionFalse(v1.VirtualMachineInstanceReady))
+
+			By("Pods should be in terminating state")
+			var pod *k8sv1.Pod
+			Eventually(func() bool {
+				pod, err = libvmi.GetPodByVirtualMachineInstance(vmi, testsuite.GetTestNamespace(vmi))
+				Expect(err).ToNot(HaveOccurred())
+				return pod.DeletionTimestamp != nil
+			}, 30*time.Second, 2*time.Second).Should(BeTrue())
+
+			By("Delete the pod to simulate that the node has been rebooted")
+			err = kubevirt.Client().CoreV1().Pods(testsuite.GetTestNamespace(vmi)).Delete(context.Background(), pod.Name,
+				metav1.DeleteOptions{GracePeriodSeconds: pointer.Int64(0)})
+			Expect(err).ToNot(HaveOccurred())
+
+			By("The controller should recreate the launcher pod and move the VMI to phase READY")
+			Eventually(ThisVMI(vmi), 200*time.Second, 2*time.Second).Should(HaveConditionTrue(v1.VirtualMachineInstanceReady))
+
+			By("Check if the VMI has been recreated, this only possible if the VMI has been on FAILED Phase")
+			fetch := ThisVMI(vmi)
```

Comment:
Replace by regular call to virtClient.VirtualMachines.Get? Looks unusual.

Reply:
Done.